contracts/Forwarder.sol

```diff
@@ -121,10 +119,10 @@ contract Forwarder is ReentrancyGuard, ERC721TokenReceiver {
     ERC721 instance = ERC721(tokenContractAddress);
     address forwarderAddress = address(this);
     address ownerAddress = instance.ownerOf(tokenId);
+    address approvedAddress=instance.getApproved(tokenId);
```

Comment:
I think if forwarder address is  not an owner but approved, we should be able to flush token to base address.
let me know your thoughts

Reply:
this is a good check, let's space out casing though just to be consistent